### PR TITLE
feat: add function name to mock creation throw error

### PR DESCRIPTION
**Pull request by distante**
# Why?

While migrating an existing project that already contains several mocks, if one of them is wrong (in my case it was a function) You have no idea which one was failing.

This PR adds a message like this:

image

PD: A linting to staged files would be cool :)

PD2: Maybe I should put the first 15 chars of the function definition instead of "Anonymous Function" ? so you get something thing like 

```invalid type: function (  () => { somecode up to 15 chars })```

### CODE REVIEW

File: libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.spec.ts [hunk 1]
```diff
@@ -34,19 +34,22 @@ describe('ts-mockito helpers', () => {
       }
     );
 
-    it('should create type and mock if type is given', () => {
```
Comment (Markus-Ende) [inline libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.spec.ts:37]: Why did you delete this test case?

File: libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.spec.ts [hunk 2]
```diff
@@ -34,19 +34,22 @@ describe('ts-mockito helpers', () => {
...
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      expect((typeAndMock.mock as any).__tsmockitoMocker.clazz).toBe(Test);
+    it('should show the name of the given function when throwing an error', () => {
+      function myTestFunction() {}
+      expect(() => createTypeAndMock(myTestFunction)).toThrowError(
+        'myTestFunction'
+      );
     });
 
+    it.each([() => {}, function () {}])(
```
Comment (Markus-Ende) [inline libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.spec.ts]: you can add the `function myTestFunction() {}` case to these test parameters here to merge the `'should show the name of the given function when throwing an error'` test case into this one (as was probably intended :smiley:)

(Please use table style test parameters like in the `'should throw error if argument is $description'` test case, then, to make it readable)

File: libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.ts [hunk 3]
```diff
@@ -19,12 +19,25 @@ export function createTypeAndMock<T>(
...
+function getFunctionNameDetail(typeOrMock: any): string {
+  if (typeof typeOrMock !== 'function') {
+    return '';
+  }
+
+  if (typeOrMock.name === '') {
+    return ' (Anonymous Function)';
+  }
+
+  return typeOrMock.name;
```
Comment (Markus-Ende) [inline libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.ts]: missing blank before the name I think?

**Review by Markus-Ende — COMMENTED** [review]
Very nice addition, thanks a bunch! :pray: 
Please see my comments for some notes.

Also, I agree with your PD2, additional infos for anonymous functions would be very cool.

File: libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.spec.ts (hunk 1 above)
Comment (distante) [reply libs/ng-mockito/ng-mockito/src/lib/ts-mockito-helpers.spec.ts:37]: This is definitily an error

**Comment by distante** [comment]
Added info about the function for anonymous functions

image
image

**Review by Markus-Ende — DISMISSED** [review]
Awesome, thanks! :heart_eyes:

**Comment by Markus-Ende** [comment]
One last thing: could you please change the scope of your commit message to `feat(ng-mockito): ...`? Its for the change log. Thank! :)

**Comment by distante** [comment]
> One last thing: could you please change the scope of your commit message to `feat(ng-mockito): ...`? Its for the change log. Thank! :)

Done!

**Review by Markus-Ende — APPROVED** [review]
